enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/hooks/use-hide-parameter.ts

```diff
@@ -0,0 +1,34 @@
+import { useSdkIframeEmbedSetupContext } from "../context";
+
+export function useHideParameter() {
+  const { settings, updateSettings } = useSdkIframeEmbedSetupContext();
+
+  const toggleParameterVisibility = (parameterName: string) => {
+    // Only dashboards supports hiding parameters
+    if (!settings.dashboardId) {
+      return;
+    }
+
+    const hiddenParameters = settings?.hiddenParameters ?? [];
+
+    const nextHiddenParameters = hiddenParameters.includes(parameterName)
```

Comment:
I will only complain a little bit since this works, but it just feels like this is not the right data structure for the task. We operate on array, and it feels like a mess removing an item and adding it back at the back of the list.

In my opinion, I would just put everything in a dictionary, that way it's really easy to toggle and very easy to check `isParameterHidden`, the code would be simpler. The only thing you need to do is to construct the `hiddenParameters` list from this dictionary.

Obviously, this is just my first thought seeing this code. You probably have more insight than I do, but what do you think about that?